src/frontends/tensorflow_common/src/op/binary_op.cpp

```diff
@@ -169,6 +171,36 @@ OutputVector translate_addv2_op(const NodeContext& node) {
     return {result};
 }
 
+OutputVector translate_equal_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"Equal"}, true);
+    auto x = node.get_input(0);
+    auto y = node.get_input(1);
+
+    auto complex_type_mark_x = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    auto complex_type_mark_y = as_type_ptr<ComplexTypeMark>(y.get_node_shared_ptr());
+
+    // both inputs are complex tensors
+    if (complex_type_mark_x && complex_type_mark_y) {
+        auto tensor1 = complex_type_mark_x->input_value(0);
+        auto tensor2 = complex_type_mark_y->input_value(0);
+
+        auto equal_op = make_shared<v1::Equal>(tensor1, tensor2);
+        // reduce along the last dimension using ReduceAnd
+        auto reduce_axes = make_shared<v0::Constant>(element::i32, Shape{1}, std::vector<int32_t>{-1});
+        auto equal_reduced = make_shared<v1::ReduceLogicalAnd>(equal_op, reduce_axes, false);
+
+        set_node_name(node.get_name(), equal_reduced);
+
+        auto complex_equal_op = make_shared<ComplexTypeMark>(equal_reduced, element::f32, element::boolean);
```

Comment:
why do we need it? Please explain.

Reply:
Apologies for the miss, I have reverted in the next revision.